### ISSUE TITLE
Storage Account name derived from Connector name

### ISSUE THREAD

**Issue by davetustin** [NetApp/terraform-provider-netapp-cloudmanager#68]
Hi,

When creating a Cloud Connector, a Storage Account is created at the same time. There is currently no way of changing the Storage Account name.

The Storage Account name uses the Cloud Connector name and appends "sa" to the end.

The Cloud Connector name needs to meet the naming conventions of a Storage Account. For example, has to be globally unique, cannot contain hyphens etc.

As an option, can a parameter be added that allows the Storage Account name to be set?

Many thanks
Dave

**Comment by davetustin** [comment]
The other issues with the Storage Account being created as part of the Connector is that there is no control over how the Storage Account is configured.

Security Centre is complaining of Secure transfer to storage accounts not being enabled, the Account kind is general purpose v1 and not v2, public access should be disallowed etc.

If the the Storage Account can be created independently of the Connector, it would be much better.

**Comment by lonico** [comment]
currently hardcoded through diagnosticsStorageAccountName
```

            "diagnosticsStorageAccountName": "[concat(toLower(parameters('virtualMachineName')),'sa')]",
            "diagnosticsStorageAccountId": "[concat('Microsoft.Storage/storageAccounts/', variables('diagnosticsStorageAccountName'))]",
            "diagnosticsStorageAccountType": "Standard_LRS",

                "dependsOn": [
                    "[concat('Microsoft.Network/networkInterfaces/', variables('networkInterfaceName'))]",
                    "[concat('Microsoft.Storage/storageAccounts/', variables('diagnosticsStorageAccountName'))]"
                ],
```

**Comment by carchi8py** [comment]
We have an internal story DEVOPS-4855 open for this.

**Comment by chuyich** [comment]
Fixed in 22.2.2
